proposals/init.md

```diff
@@ -0,0 +1,582 @@
+
+```cs
+var s = new Student()
+{
+    FirstName = "Jared",
+    LastName = "Parosns",
+};
+s.LastName = "Parsons"; // Error: LastName is not settable
+```
+
+The rules around when `init` accessors are settable extend across type
+hierarchies. If the member is accessible and the object is known to be in the
+construction phase then the member is settable. That specifically allows for
+the following:
+
+```cs
+class Base
+{
+    public bool Value { get; init; }
+}
+
+class Derived : Base
+{
+    Derived()
+    {
+        // Not allowed with get only properties but allowed with init
+        Value = true;
+    }
+}
+
+class Consumption
+{
+    void Example()
+    {
+        var d = new Derived() { Value = true; };
+    }
+}
+
+```
+
+At the point a `init` accessor is invoked the instance is known to be 
+in the open construction phase. Hence an `init` accessor is allowed to take 
+the following actions in addition to what a normal `set` accessor cand do:
+
+1. Call other `init` accessors available through `this`
+1. Assign `readonly` fields declared on the same type
+
+```cs
+class Complex
+{
+    readonly int Field1;
+    int Field2;
+    int Prop1 { get; init ; }
+    int Prop2
+    {
+        get => 42;
+        set
+        {
+            Field1 = 13; // okay
+            Field2 = 13; // okay
+            Prop1 = 13; // okay
+        }
+    }
+}
+```
+
+The ability to assign `readonly` fields from an `init` accessor is limited to 
+those fields declared on the same type as the accessor. It cannot be used to 
+assign `readonly` fields in a base type. This rule ensures that type authors
+remain in control over the mutability behavior of their type. Developers who do
+not wish to utilize `init` cannot be impacted from other types choosing to
+do so:
+
+```cs
+class Base
+{
+    internal readonly int Field;
+    internal int Property
+    {
+        get => Field;
+        init => Field = value; // Okay
+    }
+
+    internal int OtherProperty { get; init; }
+}
+
+class Derived : Base
+{
+    internal readonly int DerivedField;
+    internal int DerivedProperty
+    {
+        get => DerivedField;
+        init
+        {
+            DerivedField = 42;  // Okay
+            Property = 0;       // Okay
+            Field = 13;         // Error Field is readonly
+        }
+    }
+
+    public Derived()
+    {
+        Property = 42;  // Okay 
+        Field = 13;     // Error Field is readonly
+    }
+}
+```
+
+When `init` is used in a virtual property then all the overrides must also
+be marked as `init`. Likewise it is not possible to override a simple 
+`set` with `init`.
+
+```cs
+class Base
+{
+    public virtual int Property { get; init; }
+}
+
+class C1 : Base
+{
+    public override int Property { get; init; }
+}
+
+class C2 : Base
+{
+    // Error: Property must have init to override Base.Property
+    public override int Property { get; set; }
+}
+```
+
+An `interface` declaration can also particpate in `init` style initalization 
```

Comment:
Typo: “participate”

Reply:
Thanks! Noticed you added a lot of typo style suggestions. If you want you can always use the "Insert Suggestion" for these types of items. The upsides are I can just click to accept your suggestion and your GitHub alias ends up in the document history. 

The click to accept is great because it's much harder for me to insert a different typo when fixing the first :smile:

![image](https://user-images.githubusercontent.com/146967/79701718-30741880-8254-11ea-9803-b7606d4cd554.png)

Reply:
Yeah, sorry about that. I would have made edit suggestions if GitHub was allowing me to do so from the mobile website – I was sitting on the couch, reviewing this from my phone 😅🙈
If I knew before that I would have ended up with this many points, I maybe would have bothered to stand up and continue on my computer.
Sorry! 😅

Reply:
> Yeah, sorry about that. I would have made edit suggestions if GitHub was allowing me to do so from the mobile website – I was sitting on the couch, reviewing this from my phone

Not a problem. Mostly I wanted to make sure you were aware of the feature so you could get credit in the document history if that was interesting to you.